File: platform/test/integration/providers/jira/test_case.mjs
#!/usr/bin/env node

import path from 'node:path';
import { pathToFileURL } from 'node:url';

const args = process.argv.slice(2);
const readArg = (name, fallback) => {
  const idx = args.indexOf(name);
  return idx >= 0 ? args[idx + 1] : fallback;
};

const appEnv = readArg('--app-env', process.env.APP_ENV || 'dev');
const configDir = readArg('--config-dir', process.env.CONFIG_DIR);
if (!configDir) {
  console.error('Missing --config-dir');
  process.exit(2);
}

const platformRoot = path.resolve(configDir, '..', '..');
const importFile = async (absPath) => import(pathToFileURL(absPath).href);

const appYamlLoadMod = await importFile(path.join(platformRoot, 'polyglot/app_yaml_load/mjs/dist/index.js'));
const authConfigMod = await importFile(path.join(platformRoot, 'packages_mjs/auth_config/dist/index.js'));

const { config: appConfig } = await appYamlLoadMod.loadAppYamlConfig({ configDir, appEnv });
const providerConfig = appConfig.getNested(['providers', 'jira'], {});
const authOptions = authConfigMod.buildSdkAuthOptions(providerConfig, 'basic_email_token');
const apiKey = authConfigMod.resolveApiKey(providerConfig);
const email = authConfigMod.resolveEmail(providerConfig) ?? null;

// --- Connection test ---
let connectionTest = { skipped: true, reason: 'no health_endpoint or credentials' };
const healthEndpoint = providerConfig.health_endpoint;
const baseUrl = providerConfig.base_url;
if (healthEndpoint && apiKey && email) {
  try {
    const url = baseUrl ? `${baseUrl.replace(/\/+$/, '')}${healthEndpoint}` : healthEndpoint;
    const basicAuth = Buffer.from(`${email}:${apiKey}`).toString('base64');
    const headers = { Authorization: `Basic ${basicAuth}` };
    const res = await fetch(url, { method: 'GET', headers });
    connectionTest = { skipped: false, connected: res.ok, status: res.status };
  } catch (err) {
    connectionTest = { skipped: false, connected: false, error: err.message };
  }
}

const summary = {
  service: 'jira',
  mode: 'direct-package-import',
  mjs_package: '@internal/auth-config',
  app_yaml_initialized: true,
  app_yaml_providers_jira: providerConfig,
  resolved_config: {
    base_url: providerConfig.base_url ?? null,
    health_endpoint: providerConfig.health_endpoint ?? null,
    auth_type: authOptions?.type ?? null,
    has_api_key: apiKey != null,
    email,
  },
  connection_test: connectionTest,
  uses_app_yaml_directly: true,
  note: 'auth_config resolves jira auth (basic_email_token) from AppYamlConfig + env/context.',
};

console.log(JSON.stringify(summary, null, 2));

File: platform/test/integration/providers/jira/test_case.py
#!/usr/bin/env python3
"""Direct package test for jira provider auth config with connection check."""

from __future__ import annotations

import argparse
import base64
import json

import re
import sys
from pathlib import Path

_HERE = Path(__file__).resolve().parent
_PLATFORM = _HERE
while _PLATFORM.name != "platform" and _PLATFORM != _PLATFORM.parent:
    _PLATFORM = _PLATFORM.parent

def _inject_paths():
    """Add polyglot/packages_py source dirs to sys.path from platform pyproject.toml."""
    toml = _PLATFORM / "pyproject.toml"
    if not toml.exists():
        return
    for line in toml.read_text().splitlines():
        m = re.search(r'path\s*=\s*"([^"]+)"', line)
        if not m:
            continue
        pkg_path = (_PLATFORM / m.group(1)).resolve()
        src = pkg_path / "src"
        sys.path.insert(0, str(src if src.is_dir() else pkg_path))

_inject_paths()

import httpx

from app_yaml_load import load_app_yaml_config
from auth_config import build_sdk_auth_options, resolve_api_key, resolve_email

def _test_connection(provider_config: dict, api_key: str | None, email: str | None) -> dict:
    health_endpoint = provider_config.get("health_endpoint")
    base_url = provider_config.get("base_url")
    if not health_endpoint or not api_key or not email:
        return {"skipped": True, "reason": "no health_endpoint or credentials"}
    try:
        url = f"{base_url.rstrip('/')}{health_endpoint}" if base_url else health_endpoint
        basic_auth = base64.b64encode(f"{email}:{api_key}".encode()).decode()
        headers = {"Authorization": f"Basic {basic_auth}"}
        resp = httpx.get(url, headers=headers, timeout=10)
        return {"skipped": False, "connected": resp.is_success, "status": resp.status_code}
    except Exception as exc:
        return {"skipped": False, "connected": False, "error": str(exc)}

def main() -> int:
    parser = argparse.ArgumentParser()
    parser.add_argument("--app-env", default="dev")
    parser.add_argument("--config-dir", required=True)
    args = parser.parse_args()

    app_cfg = load_app_yaml_config(config_dir=args.config_dir, app_env=args.app_env).config
    provider_config = app_cfg.get_nested("providers", "jira", default={})
    auth_options = build_sdk_auth_options(provider_config, default_auth_type="basic_email_token")
    api_key = resolve_api_key(provider_config)
    email = resolve_email(provider_config)

    connection_test = _test_connection(provider_config, api_key, email)

    summary = {
        "service": "jira",
        "mode": "direct-package-import",
        "python_package": "auth_config",
        "app_yaml_initialized": True,
        "app_yaml_providers_jira": provider_config,
        "resolved_config": {
            "base_url": provider_config.get("base_url"),
            "health_endpoint": provider_config.get("health_endpoint"),
            "auth_type": auth_options.get("type") if auth_options else None,
            "has_api_key": api_key is not None,
            "email": email,
        },
        "connection_test": connection_test,
        "uses_app_yaml_directly": True,
        "note": "auth_config resolves jira auth (basic_email_token) from AppYamlConfig + env/context.",
    }
    # Redact sensitive fields before logging (CodeQL py/clear-text-logging-sensitive-data)
    _sensitive = {"api_key", "token", "secret", "password", "client_secret"}
    summary["app_yaml_providers_jira"] = {k: ("*******" if k in _sensitive else v) for k, v in summary["app_yaml_providers_jira"].items()}
    print(json.dumps(summary, indent=2))
    return 0

if __name__ == "__main__":
    raise SystemExit(main())
